tendermint/src/lite_impl/signed_header.rs

```diff
@@ -47,6 +47,13 @@ impl lite::Commit for block::signed_header::SignedHeader {
         if self.commit.precommits.len() != vals.validators().len() {
             return Err(lite::Error::InvalidCommitSignatures);
         }
+        // TODO: https://github.com/interchainio/tendermint-rs/pull/143/commits/0a30022fa47e909e6c7b20417dd178c8a3b84838#r374958528
+        // if let Some(parts) = &self.commit.block_id.parts {
+        //     if parts.hash != self.header_hash() {
+        //         // TODO(ismail): we need a better error here!
+        //         return Err(lite::Error::InvalidCommit);
+        //     }
+        // }
```

Comment:
I think it's OK to actually delete this now (instead of commenting it out), format this PR with `cargo fmt --` and merge this as is. I think it would be OK to fix the one failing test in a follow up PR.